pkg/controllers/workspace_controller.go

```diff
@@ -831,16 +770,52 @@ func (c *WorkspaceReconciler) applyInference(ctx context.Context, wObj *kaitov1a
 				existingObj = &appsv1.Deployment{}
 
 			}
-
+			revisionStr := wObj.Annotations[kaitov1alpha1.WorkspaceRevisionAnnotation]
 			if err = resources.GetResource(ctx, wObj.Name, wObj.Namespace, c.Client, existingObj); err == nil {
 				klog.InfoS("An inference workload already exists for workspace", "workspace", klog.KObj(wObj))
+				switch deployment := existingObj.(type) {
+				case *appsv1.Deployment:
```

Comment:
You can use the condition in line 767 instead of casting the type. 

You may also remove the comment in line 764.

Reply:
And we discussed that you only do deployment update when workspace revision number is different from deployment revision number, right?

Reply:
> And we discussed that you only do deployment update when workspace revision number is different from deployment revision number, right?

Yes, the comparison is on line 793